crates/starknet-devnet-core/src/state/mod.rs

```diff
@@ -285,8 +297,38 @@ impl CustomStateReader for StarknetState {
             || self.get_compiled_contract_class(class_hash.into()).is_ok()
     }
 
-    fn get_rpc_contract_class(&self, class_hash: &ClassHash) -> Option<&ContractClass> {
-        self.rpc_contract_classes.committed.get(class_hash)
+    fn get_rpc_contract_class(
+        &self,
+        class_hash: &ClassHash,
+        block_id: &BlockId,
+    ) -> Option<ContractClass> {
+        let class_storage = self.rpc_contract_classes.read();
```

Comment:
The if -> else if statement makes it hard to understand what is the behavior for BlockId::Pending. Also this function is public and can be called from everywhere, imo better approach is to have an enum (for example NumberOrTag) that represents the possible variants of block_id (number, tag) argument and be able to convert from BlockId to "NumberOrTag". This way you will enforce at compile time that BlockId::Hash is not provided

Reply:
Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/499/commits/40b825468d3d2ba9c6af390c765dc916e2be4215